src/hooks/usePermitAllowance.ts

```diff
@@ -81,12 +81,12 @@ export function useUpdatePermitAllowance(
       const signature = await signTypedData(provider.getSigner(account), domain, types, values)
       onPermitSignature?.({ ...permit, signature })
       return
-    } catch (e: unknown) {
+    } catch (e) {
       const symbol = token?.symbol ?? 'Token'
       if (didUserReject(e)) {
         throw new UserRejectedRequestError(`${symbol} permit allowance failed: User rejected signature`)
       }
-      throw new Error(`${symbol} permit allowance failed: ${e instanceof Error ? e.message : e}`)
+      throw new Error(`${symbol} permit allowance failed: ${e instanceof Error || e.message ? e.message : e}`)
```

Comment:
you might be better off checking `typeof e === 'object' ? e.message ?? e.reason ?? 'unknown' : e`

Reply:
you could keep the `e: unknown` typing above

Reply:
should this just be a util? I feel like this pattern occurs a lot, and we often have this same bug - of logging an object when we don't mean to.

Reply:
i like that